Makefile.pre.in

```diff
@@ -646,6 +646,20 @@ all:		@DEF_MAKE_ALL_RULE@
 # all.
 .PHONY: all
 
+# Provide quick help for common Makefile targets.
+.PHONY: help
+help:
+	@echo "Please use \`make <target>' where <target> is one of"
+	@echo "  clean      to remove build files"
```

Comment:
I suggest to repeat make, like `make clean`. So you can document `make` alone.

Reply:
I think this follows the example of the sphinx makefile in the `Doc` directory

Reply:
Sphinx Makefile has no `make` command.

Reply:
Typing `make` shows help, which follows the format also used here.

Reply:
I believe @merwok was referring to CPython's `Doc/Makefile`. That is, in fact, what gave me the idea to add a `help` target to the main Makefile.

Reply:
Nothing type `make all`. If you don't want to repeat "make", maybe document "make" outside the list?

Reply:
> Nothing type `make all`.

This statement's not parsing for me.

As for documenting `make` outside the list, the first `@echo` is

```
@echo "Please use \`make <target>' where <target> is one of"
```

Shouldn't that be enough? (I will remove the square brackets around "all." This is developer documentation for Unix-like platforms. Can't we assume some familiarity with `make` or at least the wherewithal to find tutorials or other documentation about the tool online?

Reply:
> Also added a grep command the user can execute to find all targets.

Can you implement it as a Makefile target instead? Such as `make list-targets`?

Reply:
I implemented your suggestion with just minor changes.

I thought about sorting the Makefile targets, but decided not to, as related targets are sometimes near one another.